waku/v2/protocol/waku_noise/noise.nim

```diff
@@ -81,9 +187,66 @@ type
   NoiseMalformedHandshake* = object of NoiseError
 
 
+#################################
+# Constants (supported protocols)
+#################################
+const
+
+  # The empty pre message patterns
+  EmptyPreMessagePattern: seq[PreMessagePattern] = @[]
+
+  # Supported Noise handshake patterns as defined in https://rfc.vac.dev/spec/35/#specification
+  NoiseHandshakePatterns*  = {
+    "K1K1":   HandshakePattern(name: "Noise_K1K1_25519_ChaChaPoly_SHA256",
+                               preMessagePatterns: @[PreMessagePattern(direction: D_r, tokens: @[T_s]),
+                                                     PreMessagePattern(direction: D_l, tokens: @[T_s])],
+                               messagePatterns:    @[   MessagePattern(direction: D_r, tokens: @[T_e]),
+                                                        MessagePattern(direction: D_l, tokens: @[T_e, T_ee, T_es]),
+                                                        MessagePattern(direction: D_r, tokens: @[T_se])]
+                               ),
+
+    "XK1":    HandshakePattern(name: "Noise_XK1_25519_ChaChaPoly_SHA256",
+                               preMessagePatterns: @[PreMessagePattern(direction: D_l, tokens: @[T_s])], 
+                               messagePatterns:    @[   MessagePattern(direction: D_r, tokens: @[T_e]),
+                                                        MessagePattern(direction: D_l, tokens: @[T_e, T_ee, T_es]),
+                                                        MessagePattern(direction: D_r, tokens: @[T_s, T_se])]
+                              ),
+
+    "XX":     HandshakePattern(name: "Noise_XX_25519_ChaChaPoly_SHA256",
+                               preMessagePatterns: EmptyPreMessagePattern, 
+                               messagePatterns:    @[   MessagePattern(direction: D_r, tokens: @[T_e]),
+                                                        MessagePattern(direction: D_l, tokens: @[T_e, T_ee, T_s, T_es]),
+                                                        MessagePattern(direction: D_r, tokens: @[T_s, T_se])]
+                              ),
+
+    "XXpsk0": HandshakePattern(name: "Noise_XXpsk0_25519_ChaChaPoly_SHA256",
```

Comment:
This is a custom handshake, used to immediately authenticate users based on some pre-shared key material (but not their static keys, since are exchanged again). It follows that security guarantees cannot be worse than XX, which has the same pattern except an extra pre-processing (with no data exchanged on the wire) of the `psk` information.